main/src/modules/reactiveLayer/behaviorManager/src/pointing.cpp

```diff
@@ -10,28 +10,59 @@ void Pointing::configure() {
 void Pointing::run(Bottle args/*=Bottle()*/) {
     yInfo() << "Pointing::run";
     Bottle *sensation = sensation_port_in.read();
-    
-    int id = yarp::os::Random::uniform(0, sensation->size()-1);
-    yDebug()<<"Randomly selected: "<< id; // should be random here 
+
+    int id = yarp::os::Random::uniform(0, sensation->size() - 1);
+    yDebug() << "Randomly selected: " << id; // should be random here 
     string obj_name = sensation->get(id).asList()->get(1).asString();
-    
+
+    iCub->opc->checkout();
+    yInfo() << "[pointing] : opc checkout";
+    list<Entity*> lEntities = iCub->opc->EntitiesCache();
+    string aName = "";
+    for (auto& entity : lEntities)
+    {
+        if (entity->entity_type() == "agent")
```

Comment:
Hi @sockchinglow, can you please check for `entity->m_present==1.0` as well? The agent should be present ..
Thanks!